vochain/state.go

```diff
@@ -58,15 +183,24 @@ func (e ErrHaltVochain) Unwrap() error { return e.reason }
 
 // State represents the state of the vochain application
 type State struct {
-	Store     statedb.StateDB
-	voteCache *lru.Cache
+	Store             *statedb.StateDB
+	Tx                *statedb.TreeTx
+	mainTreeViewValue atomic.Value
+	voteCache         *lru.Cache
 	ImmutableState
 	mempoolRemoveTxKeys func([][32]byte, bool)
 	txCounter           int32
 	eventListeners      []EventListener
 	height              uint32
 }
 
+// NOTE(Edu): It is my understanding that all write operations to the State
+// come from processing transaction, which are always processed serially, and
+// thus no thread-safety is required.  The arbo-based StateDB supports
+// concurrent reads to the TreeView while a new transaction is opened and
+// written serially, so there should not be need for this mutex anymore.
+// TODO: Remove the mutex and the locking once we're ready to do integration
+// tests.
```

Comment:
another note for you: when you do that, also test with `go test -race`, because it will do a better job at finding data races (e.g. concurrent non-atomic writes to the same memory address).